iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber_uds.inl

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+
+#include "owl/kom/event_subscriber_uds.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventSubscriberUds<T>::EventSubscriberUds(const core::String&,
+                                                 const core::String& instance,
+                                                 const core::String&) noexcept
+    : m_uds(std::move(iox::posix::UnixDomainSocket::create(instance, iox::posix::IpcChannelSide::SERVER)
```

Comment:
Just write:
```cxx
m_uds{iox::posix::UnixDomainSocket::create(instance, iox::posix::IpcChannelSide::SERVER).expect("Failed to create UNIX domain socket!")}
```
It does the same thing and was introduced with the functional interface PR.